research/2022-07-15-relay-anonymity.md

```diff
@@ -0,0 +1,323 @@
+This provides transparency with respect to Waku's current privacy and anonymity guarantees, and also identifies weak points that we have to address.
+
+In this post, we first give an informal description of security, privacy and anonymity in the context of Waku v2.
+For each definition, we summarize Wakus current guarantees regarding the respective property.
+We also provide attacker models, an attack-based threat model, and a first anonymity analysis of [Waku v2 relay](https://rfc.vac.dev/spec/11/) within the respective models.
+
+Waku comprises many protocols that can be combined in a modular way.
+For our privacy and anonymity analysis, we start with the relay protocol because it is at the core of Waku v2 enabling Waku's publish subscribe approach to P2P messaging.
+In its current form, Waku relay is a minor extension of [libp2p GossipSub](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/README.md).
+
+## Informal Definitions: Security, Privacy, and Anonymity
+
+The concepts of security, privacy, and anonymity are linked and have a quite a bit of overlap.
+
+### Security
+
+Of the three, [Security](https://en.wikipedia.org/wiki/Information_security) has the clearest agreed upon definition,
+at least regarding its key concepts: *confidentiality*, *integrity*, and *availability*.
+
+* confidentiality:  data is not disclosed to unauthorized entities.
+* integrity: data is not modified by unauthorized entities.
+* availability: data is available, i.e. accessible by authorized entities.
+
+While these are the key concepts, the definition of information security has been extended over time including further concepts,
+e.g. [Authentication](https://en.wikipedia.org/wiki/Authentication) and [Non-repudiation](https://en.wikipedia.org/wiki/Non-repudiation).
+We might cover these in future posts.
+
+### Privacy
+
+Privacy allows users to choose which data and information
+
+* they want to share
+* and with whom they want to share it.
+
+This includes data and information that is associated with and/or generated by users.
+Protected data also comprises metadata that might be generated without users being aware of it.
+
+Often privacy is realized by the confidentiality property of security.
+This neither makes privacy and security the same, nor the one a sub category of the other.
+While security is abstract itself (its properties can be realized in various ways), privacy lives on a more abstract level using security properties.
+Privacy typically does not use integrity and confidentiality. An adversary who has no access to the private data, because the message has been encrypted, could still alter the message.
+
+Waku is privacy-preserving. It offers:
+
+* confidentiality, because message content is only disclosed to the indented receivers.
+  We have various mechanisms in place to guarantee this, the strongest one being the [Noise Protocol Framework](https://noiseprotocol.org/).
```

Comment:
A clarification: Noise is a key-exchange mechanism to compute secrets used for encrypting data, so that is possible to have an high level of confidentiality, integrity, authentication, etc.